src/averageAges.js

```diff
@@ -56,6 +80,23 @@ function calculateWomenAverageAge(people, withChildren) {
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
   // write code here
+  const children = people.filter(person => (
+    people.some(mother => mother.name === person.mother) && (
+      (onlyWithSon)
+        ? person.sex === 'm'
+        : true
+    )
+  ));
+
+  const personAges = children.map(child => (
+    child.born - people.find(mother => mother.name === child.mother).born
+  ));
+
+  const totalAges = personAges.reduce((prevValue, currentValue) => (
+    prevValue + currentValue
+  ));
+
+  return Math.round(totalAges / personAges.length * 100) / 100;
```

Comment:
you do the same thing in every function, with calculating average age, make it a function helper, don't repeat yourself